packages/wallets/strike/src/adapter.ts

```diff
@@ -0,0 +1,181 @@
+
+    private _connecting: boolean;
+    private _wallet: StrikeWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+
+    constructor(config: StrikeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (this._wallet) {
+                    this._readyState = WalletReadyState.Loadable;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let StrikeClass: typeof StrikeWallet;
+            let wallet: StrikeWallet;
+            await import('@strike-protocols/solana-wallet-adapter').then(async (strikeSolanaWalletAdapter) => {
+                    StrikeClass = strikeSolanaWalletAdapter.StrikeWallet;
+                    try {
+                        wallet = window.strike || new StrikeClass();
+                    } catch (error: any) {
+                        throw new WalletConfigError(error?.message, error);
+                    }
+                    try {
+                        this._publicKey = await wallet.connect(this.url);
+                    } catch (error: any) {
+                        throw new WalletAccountError(error?.message, error);
+                    }
+
+                    this._wallet = wallet;
```

Comment:
The rest of this can also be simplified ala the Torus adapter. It might be helpful to just copy that and change the minimum for your SDK interface.

Reply:
sorry, not sure I'm following what needs to be simplified -- we had actually started by copying the torus adapter and changing it -- from this point on I think this file is pretty much identical to the torus adapter